uni/ulsp/database.icn

```diff
@@ -0,0 +1,117 @@
+
+# -------------------------------------------------------------------------
+# ---------------------- Database Class -----------------------------------
+# -------------------------------------------------------------------------
+#
+# Description: 
+#
+#
+#
+
+
+package ulsp
+
+import UniDoc
+
+
+class LSPDB(
+	package_db,
+	built_in_functions,           # List containing all native function names
+    action_words,                 # List containing the action words (subset of reserved words)
+    declarative_expressions,      # List containing the declarative expressions (subset of reserved words)
+    syntax_reserved_words,        # List containing the syntax reserved words (subset of reserved words)
+    keywords,                     # List containing all keywords
+    preprocessors                 # List containing all preprocessors
+)
+		
+		
+	method build()	
+		build_package_db()
+		built_in_functions := [:function():]
+		action_words := ["break", "case", "create", "critical", "every", "fail", "if", "initial", "initially", "next", "not", "repeat", "return", "suspend", "thread", "to", "until", "while"]
+		declarative_expressions := ["abstract", "class", "global", "import", "invocable", "link", "local", "method", "package", "procedure", "record", "static"]
+		syntax_reserved_words := ["all", "by", "default", "do", "end", "else", "of", "then"]
+		keywords := ["&allocated", "&ascii", "&clock", "&col", "&collections", "&column", "&control", "&cset", "&current", "&date", "&dateline", "&digits", "&dump", "&e", "&errno", "&error", "&errornumber", "&errortext", "&errorvalue", "&errout", "&eventcode", "&eventsource", "&eventvalue", "&fail", "&features", "&file", "&host", "&input", "&interval", "&lcase", "&ldrag", "&letters", "&level", "&line", "&lpress", "&lrelease", "&main", "&mdrag", "&meta", "&mpress", "&mreleasee", "&now", "&null", "&output", "&phi", "&pi", "&pick", "&pos", "&progname", "&random", "&rdrag", "&regions", "&resize", "&row", "&rpress", "&rrelease", "&shift", "&source", "&storage", "&subject", "&time", "&trace", "&ucase", "&version", "&window", "&x", "&y"]
+		preprocessors := ["$define", "$else", "$endif", "$error", "$ifdef", "$ifndef", "$include", "$line", "$undef", "#line"]
+	end
+
+    method build_package_db()
+
+	    paths := []
+	    put(paths, "/home/mark/Documents/unicon/uni/3d")
+		put(paths, "/home/mark/Documents/unicon/uni/gui")
+		#put(paths, "/home/mark/Documents/unicon/uni/parser")
+		put(paths, "/home/mark/Documents/unicon/uni/xml")
+		put(paths, "/home/mark/Documents/unicon/uni/lib")
+		put(paths, "/home/mark/Documents/unicon/ipl/procs")
+		put(paths, "/home/mark/Documents/unicon/uni/unidoc")
```

Comment:
path is still pointing at your local environment

Reply:
Paths now relative to unicon directory
Related commit: https://github.com/uniconproject/unicon/pull/305/commits/979d763d00190a6d75441114c78588f5faffa372